sigopt/xgboost/compute_metrics.py

```diff
@@ -1,19 +1,76 @@
 import numpy
-from sklearn.metrics import (
-  accuracy_score,
-  classification_report,
-  mean_absolute_error,
-  mean_squared_error
-)
+try:
+  from sklearn.metrics import (
+    accuracy_score,
+    classification_report,
+    mean_absolute_error,
+    mean_squared_error
+  )
+  HAS_SKLEARN = True
+except ImportError as e:
+  HAS_SKLEARN = False
+
+
+def compute_positives_and_negatives(y_true, y_pred, class_label):
+  tp = numpy.count_nonzero(numpy.logical_and(y_true == class_label, y_pred == class_label))
+  tn = numpy.count_nonzero(numpy.logical_and(y_true != class_label, y_pred != class_label))
+  fp = numpy.count_nonzero(numpy.logical_and(y_true != class_label, y_pred == class_label))
+  fn = numpy.count_nonzero(numpy.logical_and(y_true == class_label, y_pred != class_label))
```

Comment:
you can slightly improve this by pre-compute the `y_true == class_label` and `y_pred == class_label` and just reuse them.